src/window.py

```diff
@@ -0,0 +1,154 @@
+    __gtype_name__ = 'WorldpossibleUpdutilWindow'
+
+    _log = UpdutilLogger()
+
+    chooser_button = Gtk.Template.Child()
+    success_result_label = Gtk.Template.Child()
+    failure_result_label = Gtk.Template.Child()
+    output_label = Gtk.Template.Child()
+    output_window = Gtk.Template.Child()
+    output_buffer = Gtk.Template.Child()
+    path_entry = Gtk.Template.Child()
+    save_button = Gtk.Template.Child()
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.chooser_button.connect('clicked', self.on_chooser_clicked)
+        self.save_button.connect('clicked', self.on_save_clicked)
+        self.show_warning_dialog()
+
+    def show_warning_dialog(self):
+        dialog = Gtk.MessageDialog(self,
+                                   Gtk.DialogFlags.MODAL,
+                                   Gtk.MessageType.INFO,
+                                   Gtk.ButtonsType.OK,
+                                   'This utility allows you to execute scripts with administrator privileges!')
+        dialog.format_secondary_text('Only choose files from trusted sources to avoid harming your computer.')
+        dialog.run()
+        dialog.destroy()
+
+    def reset(self):
+        self.success_result_label.hide()
+        self.success_result_label.set_text('')
+        self.failure_result_label.hide()
+        self.failure_result_label.set_text('')
+        self.output_label.hide()
+        self.output_window.hide()
+        self.output_buffer.set_text('')
+        self.path_entry.set_text('')
+        self.save_button.hide()
+
+    def on_chooser_clicked(self, button):
+        self.reset()
+
+        chooser = Gtk.FileChooserNative.new('Choose Script', self,
+                                            Gtk.FileChooserAction.OPEN,
+                                            'Run', None)
+        res = chooser.run()
+        if res != Gtk.ResponseType.ACCEPT:
+            return
+
+        path = chooser.get_filename()
+        self.path_entry.set_text(path)
+
+        if not os.access(path, os.X_OK):
+            self.failure_result_label.set_text('Error executing script: executable permission not set')
+            self.failure_result_label.show()
+            return
+
+        self.chooser_button.set_sensitive(False)
+        self._log.info('Executing script on host system: {}'.format(path))
+        if os.path.isfile('/.flatpak-info'):
+            popen_args = ['flatpak-spawn', '--host', 'pkexec', path, None]
+        else:
+            popen_args = ['pkexec', path, None]
+
+        try:
+            proc = Gio.Subprocess.new(popen_args,
+                                      Gio.SubprocessFlags.STDOUT_PIPE |
+                                      Gio.SubprocessFlags.STDERR_MERGE)
+            proc.communicate_utf8_async(None, None, self.on_process_exit, None)
+        except GLib.Error as err:
+            self.failure_result_label.set_text('Error executing script: {}'.format(err.message))
```

Comment:
In general, you might enjoy making friends with [f-strings](https://docs.python.org/3/reference/lexical_analysis.html#f-strings)

```suggestion
            self.failure_result_label.set_text(f'Error executing script: {err.message}')
```

Reply:
Looks neat, but probably better to stick to the older way in case someone tries to use this on an old OS outside Flatpak.